static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -141,8 +148,55 @@ function FlamegraphZoomView({
 
   useEffect(() => {
     const onKeyDown = (evt: KeyboardEvent) => {
+      if (!flamegraphRenderer) {
+        return;
+      }
+
       if (evt.key === 'z' && evt.metaKey) {
-        dispatchFlamegraphState({type: evt.shiftKey ? 'redo' : 'undo'});
+        const action = evt.shiftKey ? 'redo' : 'undo';
+
+        if (action === 'undo') {
+          const previousPosition = peekHistory()?.position.view;
```

Comment:
Early returning here is annoying because we always want to dispatchFlameGraphState from L199, this is more of a "side effect"